### PR TITLE
docs(test13): hardfork RC series, master-based (review-only against `master`)

### PR DESCRIPTION
**Pull request by aeddi**
**Not meant to be merged.** This PR is opened against `master` purely so reviewers can use GitHub's compare view — it gives a single click-through diff for every rc-master branch in the series and an easy "Files changed" tab. The actual work lands by cherry-picking from the rc branches into their respective upstream PRs (or via a coordinated hardfork from `master` directly). Don't press the green button.

A stacked branch series that carries the gnoland1 → test-13 hardfork work, **rebased onto `master`** instead of `chain/gnoland1`. Each rc-master is a strict superset of the previous one — branch it off, add a focused delta, keep the older rcs intact so bisecting and reviewing never lose history. Open this PR against the highest rc-master ([`chain/test13-rc5-master`](https://github.com/aeddi/gno/tree/chain/test13-rc5-master)) unless you're reviewing one specific layer in isolation.

This stack is the master-based companion to [PR #5589](https://github.com/gnolang/gno/pull/5589) (the chain/gnoland1-based variant). Both produce a bootable test-13 genesis from the same gnoland1 history; this one inherits everything master has merged since `chain/gnoland1` was cut — notably the gas-storage refactor ([#5415](https://github.com/gnolang/gno/pull/5415)), the gas-model recalibration ([#5291](https://github.com/gnolang/gno/pull/5291)), type-persistence dedup ([#5544](https://github.com/gnolang/gno/pull/5544)), bptree ([#5475](https://github.com/gnolang/gno/pull/5475)), account sessions ([#5307](https://github.com/gnolang/gno/pull/5307)), and **PR [#5485](https://github.com/gnolang/gno/pull/5485) (valset-via-params v3) which has now landed on master** — at the cost of one extra layer of launch-readiness fixes that the older base didn't need.

A design/rationale writeup for what's specific to the master-basing lives in the ADR: [`gno.land/adr/pr5597_test13_hardfork_launch_master_based.md`](https://github.com/aeddi/gno/blob/636a8433175f0d83068faa8105bbe497861f3dfe/gno.land/adr/pr5597_test13_hardfork_launch_master_based.md). The full launch playbook — migration sequence 01 → 07, replay posture, intentional divergences, launch verification — is mostly identical to #5589's ADR; the new one only documents what's different.

> **Status (post `#5485` integration)**: validated end-to-end — `make smoketest` clean (0 failures, was 44 before the storage-deposit fix), 2-node cluster boots and advances past the genesis block in lockstep with peers connected, `make assert-migrations` 9 of 9 ok, `make verify-reproducibility` byte-identical SHA across two clean builds, keyless validator mode (gnokms-style: `VALIDATOR_ADDRESS` + `VALIDATOR_PUBKEY`) produces a bootable genesis without a `priv_validator_key.json` on disk. See ADR §"Status" for the full validation matrix.

## Branch graph

Each row links to the branch tree and shows the cumulative diff against the immediate predecessor (what this rc-master adds).

> [`master`](https://github.com/gnolang/gno/tree/master)
> ↓
> [`chain/test13-base-master`](https://github.com/aeddi/gno/tree/chain/test13-base-master) ([diff vs `master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-base-master)) — hf-glue testbed (PR #5486) merged with master + the two test13-specific commits that aren't already on master
> ↓
> [`chain/test13-rc1-master`](https://github.com/aeddi/gno/tree/chain/test13-rc1-master) ([diff vs `chain/test13-base-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-base-master...aeddi:gno:chain/test13-rc1-master)) — tier-1 audits (migrations, tx export, state, balances)
> ↓
> [`chain/test13-rc2-master`](https://github.com/aeddi/gno/tree/chain/test13-rc2-master) ([diff vs `chain/test13-rc1-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc1-master...aeddi:gno:chain/test13-rc2-master)) — tier-2 resilience (reproducibility, nil-mpkg, validator ops, halt)
> ↓
> [`chain/test13-rc3-master`](https://github.com/aeddi/gno/tree/chain/test13-rc3-master) ([diff vs `chain/test13-rc2-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc2-master...aeddi:gno:chain/test13-rc3-master)) — tier-3 audits (realm imports, gas modes, repro-doc fix)
> ↓
> [`chain/test13-rc4-master`](https://github.com/aeddi/gno/tree/chain/test13-rc4-master) ([diff vs `chain/test13-rc3-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc3-master...aeddi:gno:chain/test13-rc4-master)) — test13-specific deploy plumbing for `r/sys/validators/v3` (the realm itself now lives on master via merged #5485)
> ↓
> [`chain/test13-rc5-master`](https://github.com/aeddi/gno/tree/chain/test13-rc5-master) ([diff vs `chain/test13-rc4-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc4-master...aeddi:gno:chain/test13-rc5-master)) — master-specific launch-readiness fixes + #5485-integration adapters + this stack's ADR

## What changed vs [PR #5589](https://github.com/gnolang/gno/pull/5589)'s rc series

- **Base**: `master` instead of `chain/gnoland1`. Pulls in everything master has merged since the gnoland1 cut — including #5485, which means the v3 valset realm itself is upstream now.
- **rc1 absorbed**: every cherry-pick on `chain/test13-rc1` has independently landed on master. There is no `rc1-master` for that purpose.
- **Renumbering**: `rc1-master` → `rc3-master` correspond to `rc4` → `rc6` from #5589 (the audit/resilience/ops layers). `rc4-master` is now what's left of the v3 deploy plumbing after #5485 landed (was the v3 cherry-pick + plumbing in earlier revisions of this PR).
- **rc5-master**: launch-readiness fixes that the chain/gnoland1 base didn't need (because it predates the gas-storage refactor) plus a handful of #5485-integration adapters surfaced by the master rebase. See ADR §1.
- **Migration sequence shrinks 01 → 08 to 01 → 07**: master's #5485 removed the configurable `vm:p:valset_realm_path` field — the realm path is hardcoded in `r/sys/params/valset.gno` instead — so the dedicated migration step that wrote it is gone. The rest of the migration sequence (valset reset, T1 rotation, sysnames flip, v3 addpkg) is unchanged.

## [chain/test13-base-master](https://github.com/aeddi/gno/tree/chain/test13-base-master)

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-base-master)**: branches from `master`, merges in the [PR #5486](https://github.com/gnolang/gno/pull/5486) hf-glue testbed branch (rather than re-importing it as a squashed diff like #5589's `chain/test13-base` did), and adds the two test13-specific commits that weren't already either on master or on #5486:

- [a7e897e58](https://github.com/aeddi/gno/commit/a7e897e58) feat(hf-glue): add chunked tx-archive fetch script for flaky RPCs
- [4ddbcb0c4](https://github.com/aeddi/gno/commit/4ddbcb0c4) feat(deployments/gnoland-1): govDAO T1 rotation migration + repair valset-reset

Plus a single conflict-resolution merge commit ([4f92182bd](https://github.com/aeddi/gno/commit/4f92182bd)) for `master` ↔ #5486 — three files, all overlapping additions kept from both sides (`tm2/pkg/sdk/auth/keeper.go`, `tm2/pkg/sdk/auth/keeper_test.go`, `tm2/pkg/sdk/baseapp.go`), and a merge for the post-#5485 master ([9b19bf353](https://github.com/aeddi/gno/commit/9b19bf353)) — four conflicts in `gno.land/pkg/gnoland/{app.go,app_test.go,mock_test.go,node_params.go}` resolved as union merges of the hardfork-replay machinery and the new params-keeper-driven valset architecture.

The rest of the diff (~200 commits) is what's on master since `chain/gnoland1` was cut and what's on #5486 since its base.

## [chain/test13-rc1-master](https://github.com/aeddi/gno/tree/chain/test13-rc1-master)

**Delta vs [`chain/test13-base-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-base-master...aeddi:gno:chain/test13-rc1-master)**: same four tier-1 audit/assertion tools as #5589's `chain/test13-rc4`. None of the replay failures panic the chain (absorbed by `--skip-failing-genesis-txs`), but silent divergence could ruin a launch. These close that hole:

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-rc1-master)** (cumulative)

- [22a42fc7a](https://github.com/aeddi/gno/commit/22a42fc7a) feat(hf-glue): add `assert-migrations` script verifying post-replay state — one positive check per migration step's intended effect
- [fc2c4a030](https://github.com/aeddi/gno/commit/fc2c4a030) feat(hf-glue): add `verify-txs-jsonl` integrity check vs source-chain RPC — asserts `total_txs` + spot-checks random heights
- [5d82ddea2](https://github.com/aeddi/gno/commit/5d82ddea2) feat(hf-glue): add `state-diff` tool comparing replay vs source-chain realms — renders each realm on both sides + diffs
- [395830bba](https://github.com/aeddi/gno/commit/395830bba) feat(hf-glue): add `audit-balances` diffing per-signer ugnot source vs replay — surfaces the accounts drained by post-mainnet storage-deposit semantics

## [chain/test13-rc2-master](https://github.com/aeddi/gno/tree/chain/test13-rc2-master)

**Delta vs [`chain/test13-rc1-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc1-master...aeddi:gno:chain/test13-rc2-master)**: tier-2 resilience work from #5589's `chain/test13-rc5`. Proves the genesis rebuilds deterministically, teaches gno to survive mid-write process kills without crash-looping, adds every remaining validator-ops primitive beyond add, and patches one missing corner in the rc1 assertion script.

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-rc2-master)** (cumulative)

- [6b0783b35](https://github.com/aeddi/gno/commit/6b0783b35) feat(hf-glue): add `verify-reproducibility` building genesis twice, asserting SHA match — local proxy for the cross-machine attestation validators will run at launch
- [24e7d7f94](https://github.com/aeddi/gno/commit/24e7d7f94) fix(gnovm): survive partial mempackage writes on restart — defensive nil-skip in `PreprocessAllFilesAndSaveBlockNodes` so a half-persisted store boots instead of crash-looping (rc5-master adds the producer-side body-first ordering that complements this)
- [dc68c9bdd](https://github.com/aeddi/gno/commit/dc68c9bdd) feat(deployments/test13.gno.land): add `rm/change-power/batch` govDAO scripts — full valset-ops surface against v3
- [4641a41a8](https://github.com/aeddi/gno/commit/4641a41a8) fix(hf-glue): accept empty string as "no pending update" in `assert-migrations`

## [chain/test13-rc3-master](https://github.com/aeddi/gno/tree/chain/test13-rc3-master)

**Delta vs [`chain/test13-rc2-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc2-master...aeddi:gno:chain/test13-rc3-master)**: tier-3 audits and a documentation fix for a debugging pitfall, plus the cherry-pick of #5589's launch ADR for shared context.

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-rc3-master)** (cumulative)

- [a968c59e4](https://github.com/aeddi/gno/commit/a968c59e4) feat(hf-glue): add `audit-realm-imports` flagging dangling imports post-fork — scans every addpkg tx (historical + genesis-mode) for imports that no longer resolve against the current stdlib + examples tree
- [4ce8766fa](https://github.com/aeddi/gno/commit/4ce8766fa) feat(hf-glue): add `compare-gas-modes` A/B-testing strict vs source replay
- [999a9fcdf](https://github.com/aeddi/gno/commit/999a9fcdf) docs(hf-glue): clarify `verify-reproducibility` assumes a clean OUT dir
- [a2dccbe2a](https://github.com/aeddi/gno/commit/a2dccbe2a) docs(test13): add gnoland1 hardfork launch ADR ([`pr5589_test13_hardfork_launch.md`](https://github.com/aeddi/gno/blob/d202bbb6768184f20c1cbbc5bd305b8095d99bf8/gno.land/adr/pr5589_test13_hardfork_launch.md))

## [chain/test13-rc4-master](https://github.com/aeddi/gno/tree/chain/test13-rc4-master)

**Delta vs [`chain/test13-rc3-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc3-master...aeddi:gno:chain/test13-rc4-master)**: test13-specific deploy + migration plumbing for `r/sys/validators/v3`. The realm itself now lives on master via merged [#5485](https://github.com/gnolang/gno/pull/5485), so what remains here is what mainnet doesn't have yet — the addpkg of v3 wrapped in a sysnames-permission window, plus the operator-facing govDAO script.

When this layer was first proposed (earlier revisions of this PR) it carried the cherry-pick of #5485 itself plus a `vm:p:valset_realm_path` migration step plus an eager-eval bug fix; **all three dropped out cleanly when #5485 landed on master** — the realm is upstream, its `NewProposalRequest` already evaluates eagerly, and the configurable realm-path field was replaced with a hardcoded constant in `r/sys/params/valset.gno` (so the migration step that wrote the param became a no-op against a removed field).

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-rc4-master)** (cumulative)

- [1444441f7](https://github.com/aeddi/gno/commit/1444441f7) feat(deployments/gnoland-1): addpkg `r/sys/validators/v3` in post-fork migration — wires migration step 06 against the upstream-merged v3 source under `examples/gno.land/r/sys/validators/v3`
- [ff79b0f64](https://github.com/aeddi/gno/commit/ff79b0f64) fix(deployments/gnoland-1): wrap v3 addpkg with sysnames-check disable/restore — gnoland1's `r/sys/names.enabled = true` at halt rejects any addpkg under the `sys` namespace; migration steps 05/07 govDAO-flip `vm:p:sysnames_pkgpath` to `""` and back around step 06 to permit one-shot deployment without permanently weakening the namespace check
- [985295c1d](https://github.com/aeddi/gno/commit/985295c1d) feat(deployments/test13.gno.land): add `add-validator.sh` targeting `r/sys/validators/v3` — operator script to grow the post-fork valset via govDAO
- [9c81af5ae](https://github.com/aeddi/gno/commit/9c81af5ae) fix(test13.gno.land/add-validator.sh): use renamed v3 `NewProposalRequest` API — adapts to master's `valr.NewProposalRequest(fn, title, description)` signature (the cherry-pick used the older `NewValsetChangeExecutor` + `dao.NewProposalRequest` two-call pattern)

## [chain/test13-rc5-master](https://github.com/aeddi/gno/tree/chain/test13-rc5-master)

**Delta vs [`chain/test13-rc4-master`](https://github.com/gnolang/gno/compare/aeddi:gno:chain/test13-rc4-master...aeddi:gno:chain/test13-rc5-master)**: master-specific launch-readiness fixes + the #5485-integration adapter commits surfaced by the master rebase. The chain/gnoland1 base predates the gas-storage refactor ([#5415](https://github.com/gnolang/gno/pull/5415)) and the partial-mpkg write-amplification pattern that rc5-master closes — none of these are needed in #5589's stack but they are required for the master stack to boot end-to-end without manual JSON patching. See [ADR §1](https://github.com/aeddi/gno/blob/636a8433175f0d83068faa8105bbe497861f3dfe/gno.land/adr/pr5597_test13_hardfork_launch_master_based.md) for the full context.

**Delta vs [`master`](https://github.com/gnolang/gno/compare/master...aeddi:gno:chain/test13-rc5-master)** (cumulative)

Genesis assembly fixes:

- [d4a163644](https://github.com/aeddi/gno/commit/d4a163644) fix(gnogenesis): default gas-storage params and `gas_replay_mode` in hardfork genesis — `buildHardforkGenesis` populates the seven post-#5415 `vm.params` fields from `vm.DefaultParams()` when the source has them all at zero (the pre-refactor signature) and sets `gas_replay_mode = "source"` when unset. Operator overrides preserved. 4 unit tests.
- [311b3971c](https://github.com/aeddi/gno/commit/311b3971c) feat(gnogenesis): add `--skip-failing-genesis-txs` and `--skip-genesis-sig-verification` flags to `fork test` — `make smoketest` now matches what production validators actually run
- [ffe5adee9](https://github.com/aeddi/gno/commit/ffe5adee9) fix(gnogenesis): top up patched-addpkg creator balance for storage deposit — `patchGenesisModeAddPkg` rewrites a genesis-mode addpkg in place; when the new files are larger than the original (master's `r/sys/params` grew by `valset.gno` from #5485), the original creator's balance — sized for the smaller original — falls short of `realm_bytes × StoragePrice` and the deploy fails with "insufficient coins", silently cascading to every realm that depends on `r/sys/params`. Fix tops up the creator's genesis balance to a conservative upper bound.
- [c3c1800d1](https://github.com/aeddi/gno/commit/c3c1800d1) fix(gnogenesis): read `storage_price` from genesis params + add top-up regression tests — replaces the previous commit's hardcoded 100 ugnot/byte with a lookup of `vm.Params.StoragePrice`, with the same fallback for gnoland1's pre-#5415 schema. 6 sub-tests covering all the edge cases.

Migration / replay fixes:

- [6c4fbdb89](https://github.com/aeddi/gno/commit/6c4fbdb89) fix(deployments/gnoland-1): harden `01_reset_valset` against v2 state drift — proposal callback consults live `valr.IsValidator(addr)` at execution time and only emits removals for validators actually present; one missing entry no longer panics the whole batch
- [1bab5ca59](https://github.com/aeddi/gno/commit/1bab5ca59) fix(gnovm/store): body-first `AddMemPackage` ordering + skip-don't-panic in `IterMemPackage` — write order is iavlStore body → baseStore index → counter, so a SIGKILL between any two writes is recoverable; `IterMemPackage` yields nil on inconsistency rather than panicking, complementing the consumer-side skip from rc2-master. Side effect: replay walltime drops from ~12 min to ~36 s. 3 unit tests.
- [272dc5733](https://github.com/aeddi/gno/commit/272dc5733) fix(deployments/gnoland-1): use gno-builtin `address` type in `01_reset_valset` (no `std` import) — matches the pattern used by the sibling `04_withdraw_*` template; importing `"std"` for `std.ParseAddress` doesn't type-check in genesis-mode `MsgRun` context

#5485-integration adapters (master-rebase fallout):

- [69a474762](https://github.com/aeddi/gno/commit/69a474762) fix(test13): adapt scripts/ADR to renamed v3 API and removed migration 08 — updates `rm-validator.sh`, `change-power.sh`, `batch-change.sh`, `assert-migrations.sh`, and the rc5-master ADR to reference `valr.NewProposalRequest(fn, title, desc)` and `node:valset:dirty` (the old v3 cherry-pick used `NewValsetChangeExecutor` + `vm:gno.land/r/sys/validators/v3:new_updates_available`)
- [37a93e295](https://github.com/aeddi/gno/commit/37a93e295) fix(test13): adapt build.sh/init-node.sh/Makefile to master gnokey/gnoland CLI changes — master flipped `gnokey maketx --broadcast` default to true, requires `--insecure-password-stdin` even for the unencrypted ephemeral key, and `gnoland config init` no longer overwrites without `-force`. Without these, `make migrate` and `make init` both abort.
- [cd5b167fc](https://github.com/aeddi/gno/commit/cd5b167fc) fix(hf-glue): genesis target alias + state-diff normaliser for master stack traces — adds `genesis: migrate` alias for `verify-reproducibility.sh`, and 14 normaliser rules covering master's `/gnoroot/`, `/usr/local/go/src/`, `/root/.cache/go-build/` paths plus `:line` numbers in panic traces (otherwise `state-diff` reports cosmetic divergences as semantic ones)

Source-chain compatibility (live test against `gno.land`):

- [8c4b8737e](https://github.com/aeddi/gno/commit/8c4b8737e) fix(contribs/tx-archive): handle `GnoSessionAccount` wrapper in account decode — `gno.land` v1.0 wraps `BaseAccount` in a `GnoSessionAccount` struct (added by #5307); without the wrapper-aware decode, every `tx-archive` fetch from production fails

Operator surface:

- [81ad16295](https://github.com/aeddi/gno/commit/81ad16295) fix(deployments/gnoland-1): also flip `syscla_pkgpath` in 05/07 disable/restore wrap — at higher halt heights, manfred's CLA signature for the then-current `r/sys/cla` hash may no longer be valid, so the v3 addpkg by manfred fails on `checkCLASignature`. Migration step 05 now also clears `vm:p:syscla_pkgpath`; step 07 restores it
- [5cefecea5](https://github.com/aeddi/gno/commit/5cefecea5) feat(hf-glue): keyless validator mode via `VALIDATOR_ADDRESS` / `VALIDATOR_PUBKEY` for remote-signer setups — production validators run gnokms-backed Secp256k1 keys (no `priv_validator_key.json` on disk); keyless mode lets `make migrate` build the migrations.jsonl (valset reset + T1 rotation) using `--address` and `--pubkey` flags directly, with cross-check that the pubkey derives to the supplied address

Documentation:

- [6f4539626](https://github.com/aeddi/gno/commit/6f4539626) docs(test13): add master-based hardfork launch ADR ([`pr5597_test13_hardfork_launch_master_based.md`](https://github.com/aeddi/gno/blob/636a8433175f0d83068faa8105bbe497861f3dfe/gno.land/adr/pr5597_test13_hardfork_launch_master_based.md)) — companion to #5589's ADR, documents what's specific to master-basing
- [8a37f4e60](https://github.com/aeddi/gno/commit/8a37f4e60) docs(test13): document storage-deposit fix and validation results in pr5597 ADR — adds the deposit fix to §1, the storage-deposit-handling-at-genesis follow-up to §4 ("what belongs upstream"), corrects the smoketest baseline to 0 failures (was ~2580 absorbed before the deposit fix), and replaces the placeholder Status section with the full validation matrix

---

## How to review

- If you want the conceptual launch playbook (migration sequence, replay posture, divergences), read [#5589's ADR](https://github.com/aeddi/gno/blob/d202bbb6768184f20c1cbbc5bd305b8095d99bf8/gno.land/adr/pr5589_test13_hardfork_launch.md). Most of it applies to this stack unchanged; the master-specific deltas (migration 01 → 07 instead of 01 → 08, smoketest baseline, etc.) are in the [master-based ADR](https://github.com/aeddi/gno/blob/636a8433175f0d83068faa8105bbe497861f3dfe/gno.land/adr/pr5597_test13_hardfork_launch_master_based.md).
- For the launch fixes themselves, **rc5-master** is the only delta that doesn't have a chain/gnoland1-stack analog. The other rcs are direct rebases.
- For the v3 valset deploy plumbing — **rc4-master** is one self-contained layer (#5589 splits it across rc2 + rc3, and the realm cherry-pick that was here in earlier revisions is now upstream).
- For the audit/tooling surface that would gate a production launch — **rc1-master + rc2-master + rc3-master**, same content as #5589's rc4 + rc5 + rc6.

### CODE REVIEW

**Comment by Gno2D2** [comment]
#### 🛠 PR Checks Summary
 🔴 Must not contain the "don't merge" label


##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)
- [ ] Determine if infra needs to be updated before merging


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 Must not contain the "don't merge" label


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: aeddi/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Must not contain the "don't merge" label</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 A label matches this pattern: don't merge (label: don't merge)

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 On no pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

<details><summary>Determine if infra needs to be updated before merging</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Or
        ├── 🔴 A changed file matches this pattern: Dockerfile
        ├── 🟢 A changed file matches this pattern: ^misc/deployments (filename: misc/deployments/gnoland-1/.gitignore)
        ├── 🔴 A changed file matches this pattern: ^misc/docker-
        └── 🔴 A changed file matches this pattern: ^.github/workflows/release.*\.yml$

```
### Can be checked by
 - team devops

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

File: docs/resources/gnoland-networks.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 | Network           | RPC Endpoint                             |  Chain ID    |
 |-------------------|------------------------------------------|--------------|
-| Betanet (current) | https://rpc.gno.land:443                 | `gnoland1`   |
+| Betanet (current) | https://rpc.gno.land:443                 | `gnoland-1`  |
 | Staging           | https://rpc.staging.gno.land:443         | `staging`    |
 | Test11            | https://rpc.test11.testnets.gno.land:443 | `test11`     |
```
Comment (moul) [inline docs/resources/gnoland-networks.md:9]:
```suggestion
| Betanet (current) | https://rpc.gno.land:443                 | `gnoland1`  |
| Staging           | https://rpc.staging.gno.land:443         | `staging`    |
| Test13            | https://rpc.test13.testnets.gno.land:443 | `test-13`     |
```

File: examples/gno.land/r/sys/names/verifier.gno [hunk 2]
```diff
@@ -2,10 +2,10 @@
 // Only address-prefix (PA) namespaces are allowed.
 package names
 
-import "chain/runtime"
+import "gno.land/p/nt/ownable/v0"
 
 var (
-	admin   = address("g1rp7cmetn27eqlpjpc4vuusf8kaj746tysc0qgh") // govdao t1 multisig
+	Ownable = ownable.NewWithAddressByPrevious("g1edq4dugw0sgat4zxcw9xardvuydqf6cgleuc8p") // genesis deployer — dropped in genesis via Enable.
```
Comment (moul) [inline examples/gno.land/r/sys/names/verifier.gno:8]: It seems there is a Git rebase or merge issue; the master branch appears unchanged since this change.

**Comment by moul** [comment]
**Tentative PR splits for easier review** (work-in-progress):

| # | scope | base | status | LOC |
|---|---|---|---|---:|
| #5605 | \`fix(gnovm/store): body-first AddMemPackage ordering + fail-fast IterMemPackage\` (rc5-master \`0e423f30\`, reworked) | master | open | +208 / -40 |
| ~~#5606~~ | ~~consumer-side defensive nil-skip~~ | ~~master~~ | **closed — wrong direction**: hid corruption instead of failing loudly. Logic folded into #5605's fail-fast iterator. | — |

@aeddi original commits preserved as authorship. More splits to follow once the prerequisite infra (\`contribs/gnogenesis/internal/fork/\`, \`misc/hf-glue/\`) lands on master via #5486 or its extractions; those splits will likely stack on #5486 rather than target master directly.
